src/Microsoft.AspNetCore.BuildTools.ApiCheck/build/Common.props

```diff
@@ -0,0 +1,15 @@
+﻿<!--
+  WARNING: These targets are intended for building Microsoft's ASP.NET Core repos and are not intended
+  for use outside of Microsoft.
+-->
+<Project>
+  <!-- Set nothing if handled in project file or already done in outer build. -->
+  <PropertyGroup>
+    <EnableApiCheck Condition=" '$(EnableApiCheck)' == '' AND '$(MSBuildRuntimeType)' == 'Core' ">true</EnableApiCheck>
+    <EnableApiCheck Condition=" '$(EnableApiCheck)' == '' ">false</EnableApiCheck>
+    <ExcludePublicInternalTypes_InApiCheck Condition=" '$(ExcludePublicInternalTypes_InApiCheck)' == '' ">true</ExcludePublicInternalTypes_InApiCheck>
+
+    <!-- Hook API checks into Pack run, prior to creating the .nupkg but after build (if any). -->
+    <GenerateNuspecDependsOn Condition=" ! $(GenerateNuspecDependsOn.Contains('ApiCheck')) ">$(GenerateNuspecDependsOn);ApiCheck</GenerateNuspecDependsOn>
```

Comment:
Sure, I'll remove the `Condition`.

Don't understand your point about `$(BuildDependsOn)` versus `$(GenerateNuspecDependsOn)`. Failures in this target cause command-line builds to fail. Also, the intent is to check the APIs of assemblies we ship (aka pack), not everything we build.